### ISSUE TITLE
Add callback to `<Toast.Toggle />` `onClick`

### ISSUE THREAD

**Issue by Amystherdam** [themesberg/flowbite-react#607]
**Describe the bug**
Hey, what's up guys? I starting use flowbite-react and i trying call the onClose() event on Toast component. When X icon is clicked he close, but commands sended nothing run

**To Reproduce**
```
<Toast
  onClose={() => {
    console.log('Here');
  }}
>
```

**Expected behavior**
When X icon of Toast will clicked,  should run commands informed on block

**System information:**

- Device: Macbook Pro
- OS: MacOS Monterey
- Browser: Chrome 109.0.5414.119 (Official Build) (arm64)
- Version: 12.5

**Project information:**

- Tailwind: 3.2.4
- Flowbite: 1.6.3
- Flowbite React: 0.3.8

**Comment by tulup-conner** [comment]
Well, the method doesn't exist. This would throw an error in TypeScript.

Anyway, what you're asking for currently doesn't exist, but it certainly could. However, that would be facilitated through `<Toast.Toggle/>`, e.g., `<Toast.Toggle onClick={..} />`.

If you're able, it would be great if you could make a PR that introduces this change.

**Comment by JackRobards** [comment]
I made a small PR to try and tackle this one. 

I thought about and could potentially see some `onClose` prop making sense on the `Toast` component itself in the future, if more ways are added to close a Toast (like auto-closing it after a delay). 

For now though I kept it simple and just started using the `onClick` prop correctly with the `Toast.Toggle` component like you suggested. Feel free to let me know if you see any issues with my change, and I'll try to fix it soon!
